src/main/java/org/springframework/data/repository/core/support/DefaultRepositoryMethodContext.java

```diff
@@ -46,15 +50,41 @@ class DefaultRepositoryMethodContext implements RepositoryMethodContext {
 		this.method = method;
 	}
 
+	/**
+	 * Creates a new {@link RepositoryMethodContext} for the given {@link Method}.
+	 *
+	 * @param method must not be {@literal null}.
+	 * @return will never be {@literal null}.
+	 */
+	public static RepositoryMethodContext forMethod(Method method) {
+
+		Assert.notNull(method, "Method must not be null!");
+
+		return new DefaultRepositoryMethodContext(AbstractRepositoryMetadata.getMetadata(method.getDeclaringClass()),
+				method);
+	}
+
+	/**
+	 * Creates a proxy {@link RepositoryMethodContext} instance suitable for dependency injection.
+	 *
+	 * @return will never be {@literal null}.
+	 */
+	public static RepositoryMethodContext getInjectionProxy() {
```

Comment:
we'll need additional native image proxy configuration for `RepositoryMethodContext`.